app/main_dev/ledger.js

```diff
@@ -0,0 +1,75 @@
+import { createLogger } from "./logging";
+import TransportNodeHid from "@ledgerhq/hw-transport-node-hid";
+import { createTransaction } from "@ledgerhq/hw-app-btc/lib/createTransaction";
+import Btc from "@ledgerhq/hw-app-btc";
```

Comment:
ugh...

It took me quite a while to remove almost all dependencies from the main_dev layer, to reduce the impact of supply chain attacks on npm deps and this is going to bring back _a lot_ of stuff (see this graph for dependencies just for this package: https://npmgraph.js.org/?q=%40ledgerhq%2Fhw-app-btc).

Is there any way we can use a similar method to what trezor uses (contain the interfacing code in a separate iframe, with only the ledger-related dependencies loaded), such that both the main_dev and preload layers do not need to require all this stuff?

Any one of these transitive deps gets compromised, it's game over for the wallets...

Reply:
Trezor has the https://github.com/trezor/trezord-go running which makes the situation a little different I think. Can we get access to usb hid using the "iframe" method? Ledger has some other transport options, but I had trouble making them work. Is there a way to run another "node" server just for ledger? My understanding in this area is not what yours is, if you have any more precise advice it would be helpful. Thanks for reviewing!

Reply:
I'll take another whack at using WebUSB https://www.npmjs.com/package/@ledgerhq/hw-transport-webusb

Does this look like it should work well in the electron enviroment?

Reply:
Yeah, that's what I thought (using web-usb I mean).

It should work fine in electron, though we might need to bump the electron version depending on what exactly the webusb relies on.